src/charm.py

```diff
@@ -40,8 +49,28 @@ def _install(self, _):
         except snap.SnapError as e:
             self.unit.status = BlockedStatus(e.message)
 
-    def _on_glauth_relation_changed(self, _):
-        self.unit.status = MaintenanceStatus("reconfiguring glauth")
+    def _on_config_data_unavailable(self, event: ConfigDataUnavailableEvent) -> None:
+        """Handle config-data-unavailable event."""
+        # If config data is unavailable, set default config
+        glauth.create_default_config(api_port=event.api_port, ldap_port=event.ldap_port)
+
+    def _on_glauth_snap_ready(self, event: GlauthSnapReadyEvent) -> None:
+        """Handle glauth-snap-ready event."""
+        glauth.start()
+        self.unit.status = ActiveStatus("glauth active")
+
+    def _on_set_confidential_action(self, event):
+        """Handle the set-confidential action."""
+        ldbd_content = {"ldap-default-bind-dn": event.params["ldap-default-bind-dn"]}
+        lp_content = {"ldap-password": event.params["ldap-password"]}
+        ldbd_secret = self.app.add_secret(ldbd_content, label="ldap-default-bind-dn")
+        logger.debug("created secret %s", "ldap-default-bind-dn")
+        lp_secret = self.app.add_secret(lp_content, label="ldap-password")
+        logger.debug("created secret %s", "ldap-password")
```

Comment:
Just use the f-string. Same reasoning as I had in your SSSD operator pull request:

```suggestion
        logger.debug(f"created secret {ldap-password}")
```